src/routes/fetchTree.tsx

```diff
@@ -32,12 +33,10 @@ import {
 async function handleDataFetch(targetUrl: string, dispatch: Function) {
   let response; // Here because of `try` block scope.
   try {
-    response = await axios.get(targetUrl);
+    response = await axios.get(targetUrl, { maxBodyLength: maxFileSize });
   } catch {
-    const errorMessage = `We weren't able to import your tree data. Please confirm this URL is correct and publicly accessible: ${targetUrl}`;
     dispatch({
       type: ACTION_TYPES.FETCH_TREE_DATA_FAILED,
```

Comment:
When the `maxBodyLength` blocks it, do you get any kind of special error you can catch on and dispatch something about the tree size being the issue? It's not really that fetching data is the problem, but rather that the target is too big. That could be annoying for an end-user to debug: if their JSON really is public and Galago can go get it, why are they seeing an error? Being clear that the issue is size would be nice.

Reply:
I agree! I didn't see any return value in their docs that would let us do this, but I'll take another look.